File: C++/225. Implement Stack using Queues.cpp
class MyStack {
public:
    queue<int> que;
    MyStack() {

    }

    void push(int x) {
        que.push(x);
    }

    int pop() {
        int size = que.size();
        while (--size) {
            int temp = que.front(); que.pop();
            que.push(temp);
        }
        int res = que.front(); que.pop();
        return res;
    }

    int top() {
        int res = this->pop();
        que.push(res);
        return res;
    }

    bool empty() {
        return que.empty();
    }
};

/**
 * Your MyStack object will be instantiated and called as such:
 * MyStack* obj = new MyStack();
 * obj->push(x);
 * int param_2 = obj->pop();
 * int param_3 = obj->top();
 * bool param_4 = obj->empty();
 */

File: Python/225. Implement Stack using Queues.py
class MyStack:

    def __init__(self):
        self.que = []

    def push(self, x: int) -> None:
        self.que.append(x)

    def pop(self) -> int:
        size = len(self.que) -1
        while (size):
            size -= 1
            self.que.append(self.que.pop(0))
        return self.que.pop(0)

    def top(self) -> int:
        res = self.pop()
        self.que.append(res)
        return res

    def empty(self) -> bool:
        return len(self.que) == 0


# Your MyStack object will be instantiated and called as such:
# obj = MyStack()
# obj.push(x)
# param_2 = obj.pop()
# param_3 = obj.top()
# param_4 = obj.empty()